clients/src/main/java/org/apache/kafka/common/utils/ImplicitLinkedHashCollection.java

```diff
@@ -676,4 +678,18 @@ final int numSlots() {
     public Set<E> valuesSet() {
         return new ImplicitLinkedHashCollectionSetView();
     }
+
+    public void sort(Comparator<E> comparator) {
+        ArrayList<E> array = new ArrayList<>();
+        Iterator<E> iterator = iterator();
+        while (iterator.hasNext()) {
+            E e = iterator.next();
+            iterator.remove();
+            array.add(e);
+        }
+        array.sort(comparator);
+        for (E e : array) {
+            add(e);
```

Comment:
I assume we don't have an `addAll` method that has inherent perf advantages, right?

Reply:
The addAll for AbstractCollection is just a simple `for` loop.